### DISCUSSION TITLE
set value where the condition is true

### DISCUSSION THREAD

**Discussion by raybellwaves** [pydata/xarray#4951] [Q&A] [answered]
Hoping to do something like `DataArray.where(cond, cond_value=1)` i.e. where a condition is true, set it as a value with the others being nan.

```
import xarray as xr
import numpy as np

da = xr.DataArray(np.arange(25).reshape(5, 5), dims=("x", "y"))
# Ideally do
# _ = da.where(da < 4, cond_value=1)

# Can do it in a two step approach
da.where(da < 4).where(da > 3, 1)
```

**Comment by keewis** [comment, marked as answer]
wouldn't `xr.where` work:
```python
xr.where(da < 4, 1, np.nan)
```
unless you want to keep the values of `da` for `x <= 3`? For that you would probably need the 2 step approach

**Reply by dcherian** [reply]
`xr.where((da > 3) & (da < 4), 1, np.nan)`?
